### ISSUE TITLE
Support for tracking observations

### ISSUE THREAD

**Issue by piyanatk** [RadioAstronomySoftwareGroup/pyuvsim#497]
Hi. I am working with the MeerKAT/SKAO team on the commissioning of the MeerKAT extension, and we want to be able to simulate visibility from a heterogeneous array with some point sources. This is already supported by `pyuvsim`, but we want to simulate a tracking observation. Is it technically possible to add a support of this? How much work would it be? I am happy to contribute some time into this. 

One solution that I have discussed with @steven-murray is to rephase the visibility to time-dependent phase centers afterwards. Since `hera_sim` uses `pyuvsim.simsetup` to initialise `matvis` and `fftvis` simulations, this "post-simulation" step would work with them as well.

**Comment by bhazelton** [comment]
The only thing that I'm not sure phasing afterwards would handle is  if the beam is moving with the source (e.g. a dish tracking a source). If that's the case, it'd probably work best to split up the sim by time step so you can specify different beams for each time.

**Comment by piyanatk** [comment]
That is true. If the dish is tracking a point source, we will have to rotate the beam to have its peak tracking the source positions, I think. Can we do that with `UVBeam`?

**Comment by bhazelton** [comment]
It's not built into UVBeam at the moment, partly because I don't know what the right way to specify such a beam is. You can certainly update the contents of a UVBeam by hand to make it work, but I think there would need to be some discussion and thinking about how to implement something like this in UVBeam if you want it built in.

**Comment by piyanatk** [comment]
Thank you for your thought @bhazelton . I might or might not end up experimenting with the rotated beam in a UVBeam depending on what we decide to use. I will post an update here if I do.
